### PR TITLE
Fixes #2407

### LINKED ISSUE #2407: cuda runtime error (101) : invalid device ordinal
CUDA_VISIBLE_DEVICES and ddp are not compatible.

https://github.com/PyTorchLightning/pytorch-lightning/blob/25ee51bc570503f331dceecc610d0eb355e22327/pytorch_lightning/trainer/distrib_data_parallel.py#L504

the pytorch respects the CUDA_VISIBLE_DEVICES.
gpu_idx should be the same as local_rank.

If I set CUDA_VISIBLE_DEVICES to be 1. It will break.

### PR DESCRIPTION
**Pull request by williamFalcon**
Removes --gpus flag from ddp call. should also help hydra users @omry

### CODE REVIEW

**Comment by williamFalcon** [comment]
Fixed @cinjon

File: pytorch_lightning/accelerators/ddp_backend.py [hunk 1]
```diff
@@ -86,12 +86,18 @@ def spawn_ddp_children(self, model):
...
-        num_gpus = torch.cuda.device_count()
-
-        if '--gpus' in command:
-            gpu_flag_idx = command.index('--gpus')
-            command[gpu_flag_idx + 1] = f'{num_gpus}'
+        # the visible devices tell us how many GPUs we want to use.
+        # when the trainer script was called the device has already been scoped by the time
+        # code reaches this point. so, to call the scripts, we need to leave cuda visible devices alone
+        # but forward the GPUs selected via environment variables
+        gpu_ids = os.environ.get('CUDA_VISIBLE_DEVICES', '')
```
Comment (Borda) [inline pytorch_lightning/accelerators/ddp_backend.py:93]: have you tested? I guess it is not correct as `CUDA_VISIBLE_DEVICES` does the mapping on cuda level so if you have applied `export CUDA_VISIBLE_DEVICES=5,6` then `torch.cuda.device_count()` gives only 2

Comment (williamFalcon) [reply pytorch_lightning/accelerators/ddp_backend.py:93]: yes i ran myself...

if i do: --gpus '2,3' your method gave 2.... then the script called by ddp had --gpus 2 which is wrong.

**Comment by omry** [comment]
Nice, thanks.

**Comment by cinjon** [comment]
Thanks! Seems to work well 👍

**Pull request merged**